application/src/main/java/org/togetherjava/tjbot/logging/FlaggedFilter.java

```diff
@@ -0,0 +1,68 @@
+package org.togetherjava.tjbot.logging;
+
+import org.apache.logging.log4j.core.Core;
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.filter.AbstractFilter;
+import org.jetbrains.annotations.NotNull;
+
+
+/**
+ * A custom Filter for Log4j2, which only lets an event pass through if a Logging Flag is set in the
+ * environment. Intended to be used for local development for devs do not want to also run the
+ * logviewer project. No errors in console or Log should appear, if the Flag is not set and the
+ * logviewer is not running.
+ */
+@Plugin(name = "FlaggedFilter", category = Core.CATEGORY_NAME, elementType = Filter.ELEMENT_TYPE)
+public class FlaggedFilter extends AbstractFilter {
```

Comment:
final

Reply:
Cannot be final, else Mockito can't subclass it